cli/tsc/99_main_compiler.js

```diff
@@ -800,12 +801,27 @@ delete Object.prototype.__proto__;
       configFileParsingDiagnostics,
     });
 
+    const checkFiles = localOnly
+      ? rootNames
+        .filter((n) => !n.startsWith("http"))
+        .map((checkName) => {
+          const sourceFile = program.getSourceFile(checkName);
+          if (sourceFile == null) {
+            throw new Error("Could not find source file for: " + checkName);
+          }
+          return sourceFile;
+        })
+      : undefined;
     const diagnostics = [
       ...program.getConfigFileParsingDiagnostics(),
-      ...program.getSyntacticDiagnostics(),
+      ...(checkFiles == null
+        ? program.getSyntacticDiagnostics()
+        : checkFiles.map((s) => program.getSyntacticDiagnostics(s)).flat()),
       ...program.getOptionsDiagnostics(),
       ...program.getGlobalDiagnostics(),
-      ...program.getSemanticDiagnostics(),
+      ...(checkFiles == null
+        ? program.getSemanticDiagnostics()
+        : checkFiles.map((s) => program.getSemanticDiagnostics(s)).flat()),
```

Comment:
I'll update this to use `ts.sortAndDeduplicateDiagnostics` because the compiler API deos that internally in `getSemanticDiagnostics` and `getSyntacticDiagnostics` when not providing a source file.